include/boost/geometry/algorithms/detail/overlay/get_turn_info_ll.hpp

```diff
@@ -553,65 +559,75 @@ struct get_turn_info_linear_linear
 
         bool res = false;
 
-        if ( is_p_spike
-          && ( BOOST_GEOMETRY_CONDITION(is_version_touches)
-            || inters.d_info().arrival[0] == 1 ) )
+        if (is_p_spike)
         {
-            if ( BOOST_GEOMETRY_CONDITION(is_version_touches) )
+            bool output_spike = false;
+            if BOOST_GEOMETRY_CONSTEXPR (is_version_touches)
             {
                 tp.operations[0].is_collinear = true;
                 tp.operations[1].is_collinear = false;
                 tp.method = method_touch;
+
+                output_spike = true;
             }
-            else // Version == append_collinear_opposite
+            else if (inters.d_info().arrival[0] == 1) // Version == append_collinear_opposite
             {
                 tp.operations[0].is_collinear = true;
                 tp.operations[1].is_collinear = false;
 
                 BOOST_GEOMETRY_ASSERT(inters.i_info().count > 1);
-
                 base_turn_handler::assign_point(tp, method_touch_interior,
                                                 inters.i_info(), 1);
+
+                output_spike = true;
             }
 
-            tp.operations[0].operation = operation_blocked;
-            tp.operations[1].operation = operation_intersection;
-            *out++ = tp;
-            tp.operations[0].operation = operation_intersection;
-            //tp.operations[1].operation = operation_intersection;
-            *out++ = tp;
+            if (output_spike)
+            {
+                tp.operations[0].operation = operation_blocked;
+                tp.operations[1].operation = operation_intersection;
+                *out++ = tp;
+                tp.operations[0].operation = operation_intersection;
+                //tp.operations[1].operation = operation_intersection;
+                *out++ = tp;
 
-            res = true;
+                res = true;
+            }
         }
 
-        if ( is_q_spike
-          && ( BOOST_GEOMETRY_CONDITION(is_version_touches)
-            || inters.d_info().arrival[1] == 1 ) )
+        if (is_q_spike)
         {
-            if ( BOOST_GEOMETRY_CONDITION(is_version_touches) )
+            bool output_spike = false;
```

Comment:
this is also change the behavior e.g. the block 
```
            tp.operations[0].operation = operation_intersection;
            tp.operations[1].operation = operation_blocked;
            *out++ = tp;
            //tp.operations[0].operation = operation_intersection;
            tp.operations[1].operation = operation_intersection;
            *out++ = tp;
```
was executed in all cases but now only in touch or append_colinnear_opposite cases. 

Is this intentional?

Reply:
If there is any change in behavior it was not intentional. But AFAIU there is no change here. We're talking about the block below at line 607(old) or 622(new) correct? In the old version it was executed:
```
if ( is_q_spike
  && ( BOOST_GEOMETRY_CONDITION(is_version_touches)
    || inters.d_info().arrival[1] == 1 ) )
```
now it's executed if `output_spike` is `true` and this is set in block:
```
if (is_q_spike)
```
and only
```
if BOOST_GEOMETRY_CONSTEXPR (is_version_touches)
```
or
```
else if (inters.d_info().arrival[1] == 1) // Version == append_collinear_opposite
```

For convenience here is [the old version](https://github.com/boostorg/geometry/blob/eb38231d367fe0b74fe8c2b4083d94870a86ac9d/include/boost/geometry/algorithms/detail/overlay/get_turn_info_ll.hpp) and [the new version](https://github.com/boostorg/geometry/blob/f25545b58e54857f56cfe90769d9bdafe4bca731/include/boost/geometry/algorithms/detail/overlay/get_turn_info_ll.hpp).

Or am I missing something?